### PR TITLE
feat: Adding API's for copying/downloading an image

### PR DESCRIPTION
**Pull request by Khushboo-dev-cpp**

### CODE REVIEW

**Review by alaibe — APPROVED** [review]

File: lib/src/DOtherSide.cpp [hunk 1]
```diff
@@ -143,6 +143,24 @@ void dos_qapplication_clipboard_setText(const char* text)
...
+    QApplication::clipboard()->setImage(image);
+}
+
+void dos_qapplication_download_image(const char *imageSource, const char *filePath)
+{
+    QByteArray btArray =  QString(imageSource).split("base64,")[1].toUtf8();
+    QImage image;
+    QString fileL = QString(filePath).replace(QRegExp("^(file:/{2})|(qrc:/{2})|(http:/{2})"), "");
+    image.loadFromData(QByteArray::fromBase64(btArray));
+    image.save(QString(fileL) + "/unknown.png");
```
Comment (alaibe) [inline lib/src/DOtherSide.cpp:161]: Don't we choose a path for the image to be saved?

**Review by iurimatias — CHANGES_REQUESTED** [review]
when I right click and click 'download image' nothing happens

File: lib/src/DOtherSide.cpp (hunk 1 above)
Comment (Khushboo-dev-cpp) [reply lib/src/DOtherSide.cpp]: We choose that via the file dialog once the download button is pressed, basically from the MessageMenuContext -> FileDialog

**Comment by Khushboo-dev-cpp** [comment]
> when I right click and click 'download image' nothing happens

Hey @iurimatias can you tell me on what OS are you running it? can you share the logs? cause it works for me!

**Comment by Khushboo-dev-cpp** [comment]
Hey @iurimatias works now on both linux and Mac

**Pull request merged**